diskqueue.go

```diff
@@ -647,6 +652,9 @@ func (d *diskQueue) ioLoop() {
 			count++
 			// moveForward sets needSync flag if a file is removed
 			d.moveForward()
+			if d.depth == 0 && d.emptyCallback != nil {
+				d.emptyCallback()
```

Comment:
There is a risk that this callback will impact performance if this call is slow. Should it be called in a go routine?